modules/sonobiBidAdapter.js

```diff
@@ -60,8 +60,10 @@ export const spec = {
       payload.us = config.getConfig('userSync').syncsPerBidder;
     }
 
-    if (deepAccess(validBidRequests[0], 'crumbs.pubcid') || deepAccess(validBidRequests[0], 'params.hfa')) {
-      payload.hfa = deepAccess(validBidRequests[0], 'params.hfa') ? deepAccess(validBidRequests[0], 'params.hfa') : `PRE-${deepAccess(validBidRequests[0], 'crumbs.pubcid')}`;
+    if (deepAccess(validBidRequests[0], 'userId.pubcid') || deepAccess(validBidRequests[0], 'params.hfa')) {
+      payload.hfa = deepAccess(validBidRequests[0], 'params.hfa') ? validBidRequests[0].params.hfa : `PRE-${validBidRequests[0].userId.pubcid}`;
+    } else if (deepAccess(validBidRequests[0], 'crumbs.pubcid') || deepAccess(validBidRequests[0], 'params.hfa')) {
+      payload.hfa = deepAccess(validBidRequests[0], 'params.hfa') ? validBidRequests[0].params.hfa : `PRE-${validBidRequests[0].crumbs.pubcid}`;
```

Comment:
already checked/used `params.hfa`, so don't need to do that again in the `else if`.

Reply:
cleanest thing to do might be just
```
if (deepAccess(validBidRequests[0], 'params.hfa')) {
    payload.hfa = deepAccess(validBidRequests[0], 'params.hfa');
} else if (deepAccess(validBidRequests[0], 'userId.pubcid')) {
    payload.hfa = `PRE-${validBidRequests[0].userId.pubcid}`;
} else if (deepAccess(validBidRequests[0], 'crumbs.pubcid')) {
    payload.hfa = `PRE-${validBidRequests[0].crumbs.pubcid}`;
}
```